src/Compilers/Core/MSBuildTaskTests/TestUtilities/DotNetSdkTestBase.cs

```diff
@@ -185,16 +182,36 @@ public DotNetSdkTestBase(ITestOutputHelper testOutputHelper)
                 { "CSharpCoreTargetsPath", csharpCoreTargets },
                 { "VisualBasicCoreTargetsPath", visualBasicCoreTargets },
                 { "MSBuildSDKsPath", sdksDir },
-                { "DOTNET_MSBUILD_SDK_RESOLVER_SDKS_DIR", sdksDir }
+                { "DOTNET_MSBUILD_SDK_RESOLVER_SDKS_DIR", sdksDir },
+                // Disable the server until it's production ready so it doesn't cause CI flakiness
+                { "DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER", "1" }
```

Comment:
This looks like the key part of the change. Could you help me understand the motivation for the other parts of the change?

Reply:
Part of the investigation made me realize we weren't uploading the `restore.binlog` file generated in the ctor. That is also an artifact that could be necessary to solve test failures. It was an oversite on my part from the last attempt to upload everything. As such I decided to just write a helper for running msbuild that made sure we uploaded everything necessary by default. So no more mistakes.